src/snowflake/snowpark/_internal/xml_reader.py

```diff
@@ -301,6 +319,11 @@ def process_xml_range(
     tag_start_2 = f"<{tag_name} ".encode()
     closing_tag = f"</{tag_name}>".encode()
 
+    # We perform raw byte‑level scanning here because we must split the file into independent
```

Comment:
I added here in `process_xml_range` before calling `find_next_opening_tag`, which is more clear IMO. @sfc-gh-jkew